### PR TITLE
Add output-type flag to cli of ccd-js-gen tool

### LINKED ISSUE #321: [ccd-js-gen] Exposing a flag in the cli of the package `ccd-js-gen` to specify which files should be generated (`TypeScript`/`Javascript`/`TypedJavaScript` or `Everything`)
**Task description**

Using the cli of the package `ccd-js-gen` in an e.g. `TypeScript` project currently produces all files (meaning `TypeScript`/`Javascript`, and `TypedJavaScript` files).

```
ccd-js-gen --module ./my-contract.wasm.v1 --out-dir ./generated
```

https://github.com/Concordium/concordium-node-sdk-js/tree/main/packages/ccd-js-gen#example

In an e.g. `TypeScript` project, I only need the `TypeScript` files (while the other files need to be removed manually). 

Task: Expose an optional  `--output-type` option that would limit the output to only the files needed. If the  `--output-type` is not specified the cli should generate `Everything`.

e.g.

```
ccd-js-gen --module ./my-contract.wasm.v1 --out-dir ./generated --output-type TypeScript
```

https://github.com/Concordium/concordium-node-sdk-js/blob/main/packages/ccd-js-gen/src/cli.ts#L23

This can be already done when writing a script:
https://github.com/Concordium/concordium-governance-committee-voting/blob/main/apps/voting/scripts/generate-election-contract-client.ts#L33

*Add label for component and priority.*

### PR DESCRIPTION
**Pull request by DOBEN**
## Purpose

Closes #321 

## Changes

Add `--output-type` flag to cli of `ccd-js-gen` package.

### CODE REVIEW

File: packages/ccd-js-gen/package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
         "build-dev": "yarn build",
         "clean": "rm -r lib",
         "lint": "eslint src/** bin/** --cache --ext .ts,.js --max-warnings 0",
-        "lint-fix": "yarn --silent lint --fix; exit 0",
+        "lint-fix": "yarn lint --fix; exit 0",
```
Comment (abizjak) [inline packages/ccd-js-gen/package.json:15]: Why is this changed?

**Review by abizjak — APPROVED** [review]
Changelog please.

File: packages/ccd-js-gen/package.json (hunk 1 above)
Comment (DOBEN) [reply packages/ccd-js-gen/package.json:15]: It caused this error:

```Unknown Syntax Error: Command not found; did you mean one of: ...```

I kept the syntax as we have it in other repos now e.g.:
https://github.com/Concordium/concordium-dapp-examples/blob/main/dapp-template/frontend/package.json#L10

File: packages/ccd-js-gen/src/cli.ts [hunk 2]
```diff
@@ -28,12 +30,33 @@ export async function main(): Promise<void> {
...
+        .option<lib.OutputOptions>(
+            '-t, --output-type [TypeScript|JavaScript|TypedJavaScript|Everything]',
+            'The output file types for the generated code.',
+            (value: string) => {
+                switch (value) {
+                    case 'TypeScript':
+                    case 'JavaScript':
+                    case 'TypedJavaScript':
+                    case 'Everything':
+                        return value as lib.OutputOptions;
```
Comment (limemloh) [inline packages/ccd-js-gen/src/cli.ts:42]: **Suggestion:** I think typescript is smart enough for us to avoid casting here.
```suggestion
                        return value;
```

**Review by limemloh — APPROVED** [review]
Nice addition! :+1:
